extensions/amp-carousel/0.1/slides.js

```diff
@@ -87,7 +87,14 @@ export class AmpSlides extends BaseCarousel {
   }
 
   /** @override */
-  goCallback(dir, animate) {
+  goCallback(dir, animate, opt_autoplay) {
+    // if `opt_autoplay` is not set, this means a user interacted with the
+    // the carousel and we cancel any current and future autoplay request
+    if (!opt_autoplay) {
```

Comment:
I don't think the autoplay needs to change. What I think should happen is autoplay binds `#go` with `1` (etc), but the next/prev button call `#next`/`#prev`. Importantly, those can only be called from user interaction, so maybe something like `#interactionNext`.